ssh: add addKeysToAgent option

lockejan opened the pull request:
### Description

add addKeysToAgent option to ssh module

### Checklist

<!--

Please go through the following checklist before opening a non-WIP
pull-request.

Also make sure to read the guidelines found at

  https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-guidelines

-->

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

rycee commented on modules/programs/ssh.nix:
```diff
@@ -355,6 +355,16 @@ in
     };
 
+    addKeysToAgent = mkOption {
+      type = types.enum [ "yes" "no" "ask" "confirm" ];
```
Unfortunately I don't think an `enum` would work in this case since it wouldn't allow a value with a timeout, such as "1h" or "confirm 1h". If we add this option then I suspect it would have to be a plain `types.str`.

lockejan replied:
My concerns persist, but I've changed the PR as suggested. I've rebased and ran all tests again. I leave the final decision to @rycee . Feel free to close or merge.

lockejan commented:
My concerns persist, but I've changed the PR as suggested. I've rebased and ran all tests again. I leave the final decision to @rycee . Feel free to close or merge.

sumnerevans commented:
Probably needs a rebase to get the tests passing.

lockejan commented:
> Probably needs a rebase to get the tests passing.

Maybe you are right. I rebased and ran the tests again locally, this time against the flake inputs `nix develop --ignore-environment .#all`.
No errors whatsoever.
Should be fine now.

rycee commented:
Sorry about the delay! Finally round to merging this 🙂

sersorrel commented:
>  Change is backwards compatible.

unfortunately this change inscrutably breaks behaviour for anyone who set `programs.ssh.extraConfig` to `"AddKeysToAgent yes"` already, since per ssh_config(5) "for each parameter, the first obtained value will be used" and the new `programs.ssh.addKeysToAgent` inserts its `AddKeysToAgent` line before `extraConfig`.
